src/main/java/nl/tudelft/contextproject/gui/CreateScriptController.java

```diff
@@ -549,40 +293,63 @@ private void setAddButton() {
             }
         });
     }
-    
+
     /**
      * Checks if the new shot being added to the script
      * is valid.
      * 
-     * @return True if it has no errors.
+     * @return True iff it has no errors.
      */
     private boolean isValidInput() {
         boolean isValid = true;
 
         if (addCamera.getSelectionModel().isEmpty()) {
-            addCamera.setStyle("-fx-border-color: red;");
+            addCamera.setStyle(REDBORDER);
             isValid = false;
         }
 
         if (addPreset.getSelectionModel().isEmpty()) {
-            addPreset.setStyle("-fx-border-color: red;");
+            addPreset.setStyle(REDBORDER);
             isValid = false;
         }
 
         if (addDescription.getText().isEmpty()) {
-            addDescription.setStyle("-fx-border-color: red;");
+            addDescription.setStyle(REDBORDER);
             isValid = false;
         }
-        
+
         return isValid;
     }
-    
+
     /**
-     * Creates a new shot based on the users' input.
+     * Checks if a newly created shot does not create an
+     * invalid script.
      * 
-     * @param data The data already in the table.
+     * @param data The table data.
+     * @return True if the script is valid, false otherwise.
+     */
+    private boolean validateScript(ObservableList<Shot> data) {
```

Comment:
Program to an interface, not to an implementation: Change ObservableList<Shot> to List<Shot>